### DISCUSSION TITLE
Does webpack bundle/transpile npm packages differently for the /app folder vs the /pages folder?

### DISCUSSION THREAD

**Discussion by GitNoise** [vercel/next.js#73472] [Help]
### Summary

I have some code that uses an npm package that works perfectly in the `/app` folder but not in the `/pages` folder. 

Is there a difference in how webpack bundles (and/or transpiles) npm packages in the different folders? Or is there something else going on?

Error when running in `/pages`:
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".svelte" for /node_modules/layerchart/dist/components/Arc.svelte
```

(I am investigation the possibility of running Svelte based packages in Next.js)

See **Additional information** for the `next.config.js`. 

### Additional information

Next.js  version: 14.2.15

next.config.js:
```js

const path = require("path");
const { sveltePreprocess } = require("svelte-preprocess");


module.exports = {
  webpack: (config, { isServer }) => {
    // Add Svelte loader configuration
    config.module.rules.push({
      test: /\.svelte$/,
      use: [
        {
          loader: "svelte-loader",
          options: {
            preprocess: sveltePreprocess(),
            emitCss: !isServer, // Don't emit CSS for server-side rendering
          },
        },
      ],
    });

    // Resolve Svelte imports correctly (including internal paths)
    config.resolve.alias = {
      ...config.resolve.alias,
      svelte: path.resolve("node_modules", "svelte/src/runtime"),
    };

    // Ensure `.svelte` extensions are handled by Webpack
    config.resolve.extensions.push(".svelte");

    return config;
  },
};
```

Repo with code: https://github.com/GitNoise/react-svelte_layerchart

**Comment by icyJoseph** [comment]
This part of my Server Components, is a bit wishy-washy, and I've mostly gone by stuff said on this podcast, perhaps this bit is interesting for you:

- https://changelog.com/jsparty/311#transcript-74

Cool stuff though! Would love to see how far you can take it!

**Reply by GitNoise** [reply]
Ah, I see! 

So in the /app folder, Webpack uses a custom Vercel/Next.js plugin (@next/react-server-dom-webpack?) to enable React Server Components, and that might explain the difference. Interesting.

**Reply by icyJoseph** [reply]
Yeah, but exactly how everything is setup, and documentation from both Next.js and React parts is still pretty much anecdotal, unless one is willing to do code spelunking
